### ISSUE TITLE
Model fails to detect repeated partial forms of the same entity

### ISSUE THREAD

**Issue by deepanwadhwa** [urchade/GLiNER#242]
Hello team,

I am a great fan of this repository and thank you for all the wonderful work.

I want to report a quirk/bug that I have noticed. I have attempted this on two versions 0.2.3 and 0.2.17.

versions : [0.2.3, 0.2.17]
models : ["urchade/gliner_medium-v2.1" , "urchade/gliner_largev2", "numind/NuNerZero"]

Description : The model fails to detect repeated partial entity if the partial entity is preceding the full entity. 

```python

import gliner
print(gliner.__version__)
from gliner import GLiNER
medium_model = GLiNER.from_pretrained("urchade/gliner_medium-v2.1")
large_model = GLiNER.from_pretrained("urchade/gliner_largev2")

t0 = "Amy Poehler is a great comedian just like Tina Fey, but Amy is from Minnesota unlike Tina."
t1 = "John Doe is suffering from broken heart syndrome. John visited Japan. John also likes Sushi. John was born in 1924."
t2 = "John is suffering from broken heart syndrome. John Doe visited Japan. John also likes Sushi. John was born in 1924."
t3 = "Lionel Messi went surfing yesterday. Lionel also went out for coffee later."
t4 = "The Glendale Den Apartments are probably the best place to live in this area. They are also known as 'The Den' sometimes. The Den also provides a heated pool and a game room."

ts = [t0,t1,t2,t3,t4]

labels = ["person","neighborhood"]

for t in ts:
    print(t)
    medium_entities = medium_model.predict_entities(t, labels)
    for entity in medium_entities:
        print("Medium model result : ",entity["text"], "=>", entity["label"])
    print(25*"-")
    large_entities = large_model.predict_entities(t, labels)
    for entity in large_entities:
        print("Large model result : ",entity["text"], "=>", entity["label"])
    print(50*"=")

``` 
The output is the following:

```shell

0.2.17
Fetching 5 files: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 44620.26it/s]
/Users/deepanwadhwa/Documents/projects/anonify/.venv/lib/python3.13/site-packages/transformers/convert_slow_tokenizer.py:559: UserWarning: The sentencepiece tokenizer that you are converting to a fast tokenizer uses the byte fallback option which is not implemented in the fast tokenizers. In practice this means that the fast version of the tokenizer can produce unknown tokens whereas the sentencepiece version would have converted these unknown tokens into a sequence of byte tokens matching the original piece of text.
  warnings.warn(
Fetching 5 files: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 5/5 [00:00<00:00, 115864.75it/s]
Amy Poehler is a great comedian just like Tina Fey, but Amy is from Minnesota unlike Tina.
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.
Medium model result :  Amy Poehler => person
Medium model result :  Tina Fey => person
-------------------------
Asking to truncate to max_length but no maximum length is provided and the model has no predefined maximum length. Default to no truncation.
Large model result :  Amy Poehler => person
Large model result :  Tina Fey => person
==================================================
John Doe is suffering from broken heart syndrome. John visited Japan. John also likes Sushi. John was born in 1924.
Medium model result :  John Doe => person
-------------------------
Large model result :  John Doe => person
==================================================
John is suffering from broken heart syndrome. John Doe visited Japan. John also likes Sushi. John was born in 1924.
Medium model result :  John => person
Medium model result :  John => person
Medium model result :  John => person
Medium model result :  John => person
-------------------------
Large model result :  John => person
Large model result :  John Doe => person
Large model result :  John => person
Large model result :  John => person
==================================================
Lionel Messi went surfing yesterday. Lionel also went out for coffee later.
Medium model result :  Lionel Messi => person
-------------------------
Large model result :  Lionel Messi => person
==================================================
The Glendale Den Apartments are probably the best place to live in this area. They are also known as 'The Den' sometimes. The Den also provides a heated pool and a game room.
Medium model result :  The Glendale Den Apartments => neighborhood
-------------------------
Large model result :  The Den => neighborhood
==================================================


``` 

In the above output, whenever a first name(John or Amy) appears after the full name(John Doe or Amy Poehler) in the same text- the model doesn't detect the first name instance(John or Amy).

But if the full name appears after the first name, then all the instances of the name are detected correctly. 

I started debugging by putting some print statements in the model.py , and I see that the model confidently labels the full name (e.g., “John Doe”) with high logits. However, in subsequent instances, when only the first name (“John”) reappears, the model’s logits drop below our threshold(closer to zero), causing these partial mentions to be excluded from the final predictions.
